README.md
# [Decanter React](https://github.com/SU-SWS/decanter-react)

[![NPM](https://img.shields.io/npm/v/decanter-react.svg)](https://www.npmjs.com/package/decanter-react)
[![Code Style](https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=flat-square)](https://github.com/prettier/prettier)

Changelog: [CHANGELOG.md](CHANGELOG.md)

## Description

A React Component Library using Decanter V7+ and TailWindCSS

## Install

```bash
npm install --save decanter-react
```

## Usage

```jsx
import React, { Component } from "react";

import { MyComponent } from "decanter-react";
import "decanter/dist/decanter.css";

class Example extends Component {
  render() {
    return <MyComponent />;
  }
}
```

## Demo and Documentation

Please see our storybook build website at [Decanter React - Stoybook](https://decanter-react.netlify.app). It is full of wonderful tools, markup, examples, and information about the components in this package.

## Accessibility

[![WCAG Conformance 2.0 AA Badge](https://www.w3.org/WAI/wcag2AA-blue.png)](https://www.w3.org/TR/WCAG20/)

Evaluation Date: 2020-05-14
This project conforms to level AA WCAG 2.0 standards as required by the university's accessibility policy. For more information on the policy please visit: [https://ucomm.stanford.edu/policies/accessibility-policy.html](https://ucomm.stanford.edu/policies/accessibility-policy.html).

## Contributing

We encourage you to contribute to Decanter-React! Please check out the Contributing to [Decanter React guide](https://github.com/SU-SWS/decanter-react/blob/main/CONTRIBUTING.md) for guidelines about how to proceed.

## License

GPL-3.0-or-later © [Stanford Web Services](https://github.com/Stanford Web Services)


CHANGELOG.md
### Changelog

All notable changes to this project will be documented in this file. Dates are displayed in UTC.

Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).

#### [1.1.1](https://github.com/SU-SWS/decanter-react/compare/1.1.0...1.1.1)

- Version 1.1.1 [`#85`](https://github.com/SU-SWS/decanter-react/pull/85)
- ADAPT-3127: Modal content component [`#64`](https://github.com/SU-SWS/decanter-react/pull/64)
- ADAPT-000: Fixup missing dismiss text. [`#76`](https://github.com/SU-SWS/decanter-react/pull/76)
- ADAPT-000: Fixup Storybook console errors. [`#77`](https://github.com/SU-SWS/decanter-react/pull/77)
- ADAPT-3828 Adjust HeroIcon types [`#80`](https://github.com/SU-SWS/decanter-react/pull/80)

#### [1.1.0](https://github.com/SU-SWS/decanter-react/compare/1.0.8...1.1.0)

> 29 September 2021

- Adapt 3717: Hero Icon updates [`#71`](https://github.com/SU-SWS/decanter-react/pull/71)
- ADAPT-3128: Error Summary Box (and List) variation [`#70`](https://github.com/SU-SWS/decanter-react/pull/70)
- ADAPT-3096: Text Field [`#65`](https://github.com/SU-SWS/decanter-react/pull/65)
- ADAPT-2102: Hero component [`#56`](https://github.com/SU-SWS/decanter-react/pull/56)

#### [1.0.8](https://github.com/SU-SWS/decanter-react/compare/1.0.7...1.0.8)

> 18 September 2021

- ADAPT-3250: Adjustments to Tailwind configuration approach to support JIT Mode. [`#62`](https://github.com/SU-SWS/decanter-react/pull/62)
- Update GlobalFooter.js [`#61`](https://github.com/SU-SWS/decanter-react/pull/61)

#### [1.0.7](https://github.com/SU-SWS/decanter-react/compare/1.0.7-alpha.0...1.0.7)

> 16 July 2021

- Update Button propTypes [`#55`](https://github.com/SU-SWS/decanter-react/pull/55)

#### [1.0.7-alpha.0](https://github.com/SU-SWS/decanter-react/compare/1.0.6...1.0.7-alpha.0)

> 15 June 2021

- ADAPT-000: NPM Package Version Updates [`#54`](https://github.com/SU-SWS/decanter-react/pull/54)
- ADAPT-000: Remove div and setdanger in alert component story. [`#52`](https://github.com/SU-SWS/decanter-react/pull/52)

#### [1.0.6](https://github.com/SU-SWS/decanter-react/compare/1.0.5...1.0.6)

> 15 June 2021

- ADAPT-000: Poster release fixup. [`#53`](https://github.com/SU-SWS/decanter-react/pull/53)
- ADAPT-2695 Create Poster component [`#50`](https://github.com/SU-SWS/decanter-react/pull/50)

#### [1.0.5](https://github.com/SU-SWS/decanter-react/compare/1.0.5-alpha.0...1.0.5)

> 2 June 2021

- ADAPT-000: React 17 Support [`#46`](https://github.com/SU-SWS/decanter-react/pull/46)

#### [1.0.5-alpha.0](https://github.com/SU-SWS/decanter-react/compare/1.0.4...1.0.5-alpha.0)

> 2 June 2021

- ADAPT-1776 Build Card Component [`#33`](https://github.com/SU-SWS/decanter-react/pull/33)

#### [1.0.4](https://github.com/SU-SWS/decanter-react/compare/1.0.4-alpha.0...1.0.4)

> 20 May 2021

- ADAPT-000: Update README.md [`#45`](https://github.com/SU-SWS/decanter-react/pull/45)

#### [1.0.4-alpha.0](https://github.com/SU-SWS/decanter-react/compare/v1.0.2...1.0.4-alpha.0)

> 20 May 2021

- ADAPT-2517: Follow-up tweaks for auto-release action. [`#44`](https://github.com/SU-SWS/decanter-react/pull/44)
- Add PurgeCSS to Storybook build [`#31`](https://github.com/SU-SWS/decanter-react/pull/31)
- Create label.yml [`#43`](https://github.com/SU-SWS/decanter-react/pull/43)
- ADAPT-000: Update contribution documentation [`#36`](https://github.com/SU-SWS/decanter-react/pull/36)

#### [v1.0.2](https://github.com/SU-SWS/decanter-react/compare/v1.0.1...v1.0.2)

> 17 May 2021

- 1.0.2 [`#42`](https://github.com/SU-SWS/decanter-react/pull/42)
- 1.0.2 [`#41`](https://github.com/SU-SWS/decanter-react/pull/41)
- Update CHANGELOG.md [`#40`](https://github.com/SU-SWS/decanter-react/pull/40)

#### [v1.0.1](https://github.com/SU-SWS/decanter-react/compare/1.0.0...v1.0.1)

> 17 May 2021

- ADAPT-000: Update packages... again. [`#39`](https://github.com/SU-SWS/decanter-react/pull/39)
- ADAPT-000: Update packages. [`#38`](https://github.com/SU-SWS/decanter-react/pull/38)
- ADAPT-2517: Debugging (Switch decanter package to use semantic versioning) [`#37`](https://github.com/SU-SWS/decanter-react/pull/37)
- ADAPT-2157: Fix for autorelease action. [`#35`](https://github.com/SU-SWS/decanter-react/pull/35)
- ADAPT-2517: Added autorelease github action. [`#34`](https://github.com/SU-SWS/decanter-react/pull/34)
- Back to dev [`ba8099c`](https://github.com/SU-SWS/decanter-react/commit/ba8099cbb3857dd50329d18f83dd9cb8a3660879)
- ADAPT-000: Readme link updates. [`1601cf1`](https://github.com/SU-SWS/decanter-react/commit/1601cf1ce8e9e6d66d07015935027797bf4434ce)

#### 1.0.0

> 14 May 2021

- ADAPT-1771: Convert string keys to ints. [`#32`](https://github.com/SU-SWS/decanter-react/pull/32)
- ADAPT-1771: Linting setup. [`#30`](https://github.com/SU-SWS/decanter-react/pull/30)
- ADAPT-2436 Add ref to CTA Button and CTA Link components [`#28`](https://github.com/SU-SWS/decanter-react/pull/28)
- ADAPT-2296: PurgeCSS Safe Code [`#27`](https://github.com/SU-SWS/decanter-react/pull/27)
- ADAPT-2272: Alerts a11y fixes for Online Giving [`#26`](https://github.com/SU-SWS/decanter-react/pull/26)
- ADAPT-2140: Add support for forwarding refs to button component. [`#25`](https://github.com/SU-SWS/decanter-react/pull/25)
- ADAPT-000: move cnbuilder from devdependecies to dependencies [`#23`](https://github.com/SU-SWS/decanter-react/pull/23)
- ADAPT-2101: Replace CLSX with CNBuilder [`#21`](https://github.com/SU-SWS/decanter-react/pull/21)
- Adapt 1701 cta v2 [`#20`](https://github.com/SU-SWS/decanter-react/pull/20)
- ADAPT-2142: Dismiss Button as its own component and Alert tweaks [`#22`](https://github.com/SU-SWS/decanter-react/pull/22)
- package up skiplink [`#19`](https://github.com/SU-SWS/decanter-react/pull/19)
- ADAPT-2060 - Skip link component [`#17`](https://github.com/SU-SWS/decanter-react/pull/17)
- Adapt-1955 Adapt-1700 Adapt 1960 Logo Lockup, Local Footer, Heading components [`#15`](https://github.com/SU-SWS/decanter-react/pull/15)

src/Date/Date.js
import React from "react";
import PropTypes from "prop-types";
import Icon from "react-hero-icon";
import { dcnb } from "cnbuilder";
import { iconPosition } from "./Date.levers";

/**
 * Date Component
 *
 * HTML Date element
 */
export const Date = React.forwardRef(
  ({ className, position, value, label, id, onChange, ...props }) => {
    // Defaults & Variables.
    // ---------------------------------------------------------------------------
    const levers = {};

    // Levers
    // ---------------------------------------------------------------------------

    // position
    if (position && iconPosition.includes(position)) {
      switch (position) {
        case "left":
          levers.positionIcon = "su-left-0 su-ml-20";
          levers.positionWrapper = "su-pl-50";
          break;

        case "right":
          levers.positionIcon = "su-right-0 su-mr-20";
          levers.positionWrapper = "su-pr-50";
          break;

        default:
        // None.
      }
    }

    return (
      <div>
        <label htmlFor={id} className="su-text-lg su-mb-12 su-flex">
          {label}
        </label>
        <span className="su-border-b su-border-black-70 su-rounded su-inline-block su-relative">
          <input
            {...props}
            onChange={(e) => onChange(e.target.value)}
            className={dcnb(
              "su-text-18 su-p-20 su-border su-border-black-20",
              "focus:su-ring focus:su-ring-4 focus:su-ring-opacity-20 focus:su-ring-digital-blue-light",
              "focus:su-border-digital-blue focus:su-border-b-4 focus:su--mb-4 focus:su-rounded",
              levers.positionWrapper
            )}
            id={id}
            value={value}
            type="date"
            required
          />
          <Icon
            icon="calendar"
            className={dcnb(
              "su-absolute su-top-2/4 su-transform su--translate-y-2/4",
              levers.positionIcon
            )}
            type="outline"
            aria-hidden="true"
          />
        </span>
      </div>
    );
  }
);

Date.propTypes = {
  /**
   * Icon Position
   */
  position: PropTypes.oneOf(iconPosition),
  /**
   * CSS Class names.
   */
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

// Default Props.
// -----------------------------------------------------------------------------
Date.defaultProps = {
  position: "right",
};
